### PR TITLE
CreateVmDialog uses Firehose

### PR DESCRIPTION
**Pull request by mareklibra**
Depends on: #502

### CODE REVIEW

**Comment by mareklibra** [comment]
No matter this is intended for openshift/console only, there will be follow-up updating web-ui to keep its master working.

File: src/components/Wizard/CreateVmWizard/providers/VMwareImportProvider/requestedResources.js [hunk 1]
```diff
@@ -15,39 +15,39 @@ export const getVmWareProviderRequestedResources = state => {
...
         namespace,
         isList: false,
-      }),
-    };
+        prop: 'v2vvmware',
+      })
+    );
   }
 
   if (activeVcenterSecretName) {
```
Comment (suomiy) [inline src/components/Wizard/CreateVmWizard/providers/VMwareImportProvider/requestedResources.js]: this should be converted as well

File: src/components/Wizard/CreateVmWizard/providers/VMwareImportProvider/VMWareImportProvider.js [hunk 2]
```diff
@@ -161,9 +166,19 @@ export class VMWareImportProvider extends React.Component {
   }
```
Comment (suomiy) [inline src/components/Wizard/CreateVmWizard/providers/VMwareImportProvider/VMWareImportProvider.js]: and flatten activeVcenterSecret

File: src/components/Wizard/CreateVmWizard/providers/VMwareImportProvider/VMWareImportProvider.js [hunk 3]
```diff
@@ -107,7 +111,8 @@ export class VMWareImportProvider extends React.Component {
   });
 
   componentDidUpdate() {
-    const { vmwareToKubevirtOsConfigMap, v2vvmware } = this.props;
+    const v2vvmware = flatten(this.props, 'v2vvmware', {});
+    const vmwareToKubevirtOsConfigMap = flatten(this.props, 'vmwareToKubevirtOsConfigMap', undefined);
```
Comment (suomiy) [inline src/components/Wizard/CreateVmWizard/providers/VMwareImportProvider/VMWareImportProvider.js]: do we really have to specify undefined value here?

Comment (mareklibra) [reply src/components/Wizard/CreateVmWizard/providers/VMwareImportProvider/VMWareImportProvider.js]: no, leftover

File: src/components/Wizard/CreateVmWizard/providers/VMwareImportProvider/requestedResources.js (hunk 1 above)
Comment (mareklibra) [reply src/components/Wizard/CreateVmWizard/providers/VMwareImportProvider/requestedResources.js]: good catch, change introduced during rebase

**Pull request merged**